test/e2e_node/memory_eviction_test.go

```diff
@@ -147,14 +160,39 @@ var _ = framework.KubeDescribe("MemoryEviction [Slow] [Serial] [Disruptive]", fu
 				halflimit := limit / 2
 
 				// Wait for at least half of memory limit to be available
-				glog.Infof("Current available memory is: %d bytes. Waiting for at least %d bytes available.", avail, halflimit)
 				if avail >= halflimit {
-					return true
+					return nil
 				}
-
-				return false
-			}, 5*time.Minute, 5*time.Second).Should(Equal(true))
-
+				return fmt.Errorf("current available memory is: %d bytes. Expected at least %d bytes available.", avail, halflimit)
+			}, 5*time.Minute, 15*time.Second).Should(BeNil())
+
+			// Finally, try starting a new pod and wait for it to be scheduled and running.
+			// This is the final check to try to prevent interference with subsequent tests.
+			podName := "admitCheckPod"
```

Comment:
It might be worth moving this eviction test to use dynamic kubelet configs
to avoid setting high eviction thresholds, which might affect the number of
parallel tests we can run on a test node.

On Thu, Aug 25, 2016 at 11:18 AM, Vish Kannan <notifications@github.com>
wrote:

> In test/e2e_node/memory_eviction_test.go
> <https://github.com/kubernetes/kubernetes/pull/31386#discussion_r76297144>
> :
>
> >  				}
> > -
> > -				return false
> > -			}, 5*time.Minute, 5*time.Second).Should(Equal(true))
> > -
> > +				return fmt.Errorf("current available memory is: %d bytes. Expected at least %d bytes available.", avail, halflimit)
> > +			}, 5*time.Minute, 15*time.Second).Should(BeNil())
> > +
> > +			// Finally, try starting a new pod and wait for it to be scheduled and running.
> > +			// This is the final check to try to prevent interference with subsequent tests.
> > +			podName := "admitCheckPod"
>
> nit: podName := "admitBestEffortPod"
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31386/files/a9febdbff997caeab9b9917b1d70c7dd1a5ff3f5#r76297144>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvIKNet2vdqwEpNjHga0K7PSIhcRFdaks5qjdx8gaJpZM4JsjQ1>
> .
>